### ISSUE TITLE
Option to automatically add newly collected items to your watchlist

### ISSUE THREAD

**Issue by fuzeman** [trakt/Plex-Trakt-Scrobbler#370]
*originally requested here: https://github.com/trakt/Plex-Trakt-Scrobbler/issues/351#issuecomment-257767624*

**Comment by geogolem** [comment]
hmm, i just wanted to make another point regarding this feature.... I use trakt and plex for both tv shows and movies. 

here is the dilemma --> plex/trakt are doing their thing for both tv shows and movies. 

In PLEX I have two libraries - Movies and TV Shows.

When I initially started using plex-trakt and trakt, I only used it for Movies and so in the plex-trakt global filters I put Movies but not TV Shows.

Everything was working great. I eventually decided to use trakt for tv shows so I added TV shows to the comma separated list in the global filters on plex trakt.

Everything is working great. 

My collection contains movies and TV shows. My Watched list contains movies and TV Shows.. 

I use the plex watchlist to manage the movies I want to watch. i.e. my watchlist contains only movies, not tv shows. I like it this way...

so in my feature request for having newly collected items to the watchlist (if they are not already in the watchlist - not sure if that matters) ---> I want to be able to specify that this happen for the "Movies" Library and not for the "TV Shows" library .. Since I dont want any tv shows on my watchlist.

Hopefully this doesn't make things too complicated.

I also thought of suggesting a feature to have multiple instances of the plex-trakt plugin.. 

At first I thought it would be nice to have one group of settings with global filter: Movies
and an entirely different group of settings for global filter "TV Shows"

... it could be the case that one only wants to track their tv watched list (history) but not their TV collection.   right now I dont think this woudl be possible with the current plex-trakt setup. as you cannot specify different syn options for different libraries, so perhaps instead of implementing specific library filters for this new add newly collected items to watchlist feature...  there should be an entirely separate feature for having different plex-trakt options for different libraries... those two features combined would take care of the use-case I initially described in this comment.

**Comment by geogolem** [comment]
Referring to my above comment about multiple profiles; perhaps having multiple profiles would also solve this issue: https://github.com/trakt/Plex-Trakt-Scrobbler/issues/327

each profile could have its own trakt authentication PIN...

and each profile could filter to only their client... because each profile would have its own client filter list as well.

**Comment by fuzeman** [comment]
@geogolem 
> so in my feature request for ensuring newly collected items are on the watchlist ---> I want to be able to specify that this happen for the "Movies" Library and not for the "TV Shows" library .. Since I dont want any tv shows on my watchlist.
>
> Hopefully this doesn't make things too complicated.

Yeah, two separate options for movies and shows could be enough for the initial implementation.

--------------------------

Your idea about profiles sounds interesting, it could be a good way to support more complex syncing configurations.

> Again I posted this here becaus my use case directly stems from the use of the new feature described by this "issue"; however I think the best way to handle tht use case would be to create another feature for mutiple plex-trakt profiles, rather than allow specific library selection within the feature this "issue" describes itself.

Do you think this feature should just be included within the new profile system as you've described?

The problem I see is that the work required to implement profiles could mean this feature takes a lot longer to be added. Initially we could just have two separate options for movies and shows, and then integrate the feature with the new profile system at a later date..

**Comment by geogolem** [comment]
I think the option "Ensure newly collected items appear on trakt watchlist" could be implemented sooner rather than later.

I think instead of implementing it with a tickbox. I suppose you could implement it as a comma separated list where the user has to specify the libraries he wants this enabled for..

i.e. it could appear in the plex-trakt config panel as:

"Ensure newly collected items exist on trakt watchlist" [ empty textbox ]


so basically the feature would be enabled only for the libraries specified in the textbox. and the initial value of the textbox would be empty.. so effectively the feature would be disabled.

if a user put Movies in the textbox. it would only do this when adding items from their Plex Movie library.
if a user put TV Shows in the textbox. it would only do this when adding items from their plex TV Shows library

if a user put Movies,TV Shows it would do this for both the Movies and TV Shows libraries (but not any others if others exist on the plex server)..

if a user put * in the box - it would do this for all libraries for which "sync collection is enabled"

just gotta make sure this doesnt conflict with the libraries global filter. i.e. if libraries global filter indicates that a given library should not be synced and that library appears in the "ensure newly collected items exist in watchlist" textbox.. they still wont be added, since nothing is being newly collected from trakts perspective for stuff from library..

I suppose the profiles would be an entirely separate thing perhaps. 

of course --> development should proceed forward as the developers see fit.. its not really a life or death feature for me to have. I havent bothered taking a look at plex-trakt plugin's code yet.. but I might check it out at some point if i have some time.

**Comment by geogolem** [comment]
I would also like to perhaps suggest, that in the plex-trakt configuration panel Clear descriptions be added of exactly what these options do in each case and how they might interact with one another.

**Comment by geogolem** [comment]
I thought of another scenario related to this feature request..

if the newly collected item exists in my trakt "watched" list, dont add it to the trakt "watch list"..

not sure if we would wanna make this another option the user can set to adjust the behavior..
